docs/DeveloperGuide.md

```diff
@@ -671,29 +673,414 @@ testers are expected to do more *exploratory* testing.
+
+2. Adding a unique customer with all parameters specified using the command alias.
+
+    1. Prerequisites: Customer with name `Betsy Crowe` or other similar names does not already exist in the address book.
+   
+    2. Test case: `addc n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal` <br>
+       Expected: The customer is successfully added. Details of the added customer shown in the status message.
+   
+3. Adding a customer with exact same name.
+
+   1. Prerequisites: Customer with name `Betsy Crowe` already exist in the address book.
+
+   2. Test case: `addcustomer n/Betsy Crowe t/friend e/betsycrowe@duplicate.com a/Newgate Prison p/12345678 t/criminal` <br>
+      Expected: No customer is added. Error details shown in the status message. Status bar remains the same.
+
+4. Adding a customer with a [similar name](#similar) and without the optional `tag` field.
+
+    1. Prerequisites: Customer with name `Betsy Crowe` but not `Betsy crowe` already exist in the address book.
+
+    2. Test case: `addcustomer n/Betsy crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567` <br>
+       Expected: The customer is successfully added. A warning and details of the added customer shown in the status message.
+
+5. Adding a customer with duplicate tags.
+
+    1. Prerequisites: Customer with name `Yu Sutong` or other similar names does not already exist in the address book.
+
+    2. Test case: `addcustomer n/Yu Sutong t/vvip t/vvip e/su@example.com a/Newgate Prison p/12345678` <br>
+       Expected: The customer is successfully added with one of the duplicated tags ignored. Details of the added customer shown in the status message.
+
+6. Adding a customer with [similar tags](#similar).
+
+    1. Prerequisites: Customer with name `Foo Chao` or other similar names does not already exist in the address book.
+
+    2. Test case: `addcustomer n/Foo Chao t/VVIP t/vvip e/su@example.com a/69, Sembawang Road. #01-01  p/12345678` <br>
+       Expected: The customer is successfully added with both similar tags accepted. A warning and details of the added customer shown in the status message.
+
+7. Adding a customer with missing compulsory field.
+
+   1. Test case: `addcustomer n/Lim Kai Xuan e/su@example.com a/69, Sembawang Road. #01-01` <br>
+      Expected: No customer is added. Error details shown in the status message. Status bar remains the same.
+
+   2. Test case: `addcustomer n/Lim Kai Xuan e/su@example.com p/12345678` <br>
+      Expected: No customer is added. Error details shown in the status message. Status bar remains the same.
+
+### Listing all customers
+
+1. Listing all customers with or without aliasing.
+
+    1. Test case: `listcustomer` <br>
+       Expected: All customers are listed. A success message shown in the status message.
+
+    2. Test case: `listc` <br>
+       Expected: All customers are listed. A success message shown in the status message.
+
+### Finding customers by name.
```

Comment:
Consider adding case where finding customer that does not exist is reflected in status bar and customer list.